core/src/Kokkos_Graph.hpp

```diff
@@ -161,6 +161,7 @@ Graph<ExecutionSpace> create_graph(Closure&& arg_closure) {
 #include <impl/Kokkos_GraphNodeImpl.hpp>
 #include <impl/Kokkos_Default_Graph_Impl.hpp>
 #include <Cuda/Kokkos_Cuda_Graph_Impl.hpp>
+#include <HIP/Kokkos_HIP_Graph_Impl.hpp>
```

Comment:
This file doesn't have guards for `KOKKOS_ENABLE_HIP`.
```suggestion
#ifdef KOKKOS_ENABLE_HIP
#include <HIP/Kokkos_HIP_Graph_Impl.hpp>
#endif
```